### PR TITLE
Follow-up for David's previous comments

### LINKED ISSUE #599: Followup David's comment in 596
https://github.com/GoogleCloudPlatform/cloud-opensource-java/pull/596/files

### PR DESCRIPTION
**Pull request by suztomo**
Fixes #599

### CODE REVIEW

File: dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ErrorType.java [hunk 1]
```diff
@@ -22,25 +22,35 @@
   CLASS_NOT_FOUND,
 
   /**
-   * The referenced class or interface found in the class path is not compatible with the source.
+   * The referenced class or interface found in the class path is not binary-compatible with the
+   * source class.
+   *
+   * <p>An example case of breaking binary-compatibility is when a superclass changes a method to
+   * {@code final} and subclass is still overriding the method. Another example is when there is a
```
Comment (elharo) [inline dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ErrorType.java]: a subclass

File: dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ErrorType.java [hunk 2]
```diff
@@ -22,25 +22,35 @@
...
+   * method call to an interface and the interface is changed to a class.
```
Comment (elharo) [inline dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ErrorType.java]: This doesn't sound quite right.

File: dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ErrorType.java (hunk 1 above)
Comment (suztomo) [reply dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ErrorType.java]: Fixed.

File: dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ErrorType.java (hunk 2 above)
Comment (suztomo) [reply dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ErrorType.java]: I'm trying to explain "If C is not an interface, interface method resolution throws an IncompatibleClassChangeError." here
https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-5.html#jvms-5.4.3.4

File: dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ErrorType.java [hunk 3]
```diff
@@ -22,25 +22,35 @@
...
   /**
-   * The referenced class or interface found in the class path is not compatible with the source.
+   * The referenced class or interface found in the class path is not binary-compatible with the
+   * source class.
+   *
+   * <p>An example case of breaking binary-compatibility is when a superclass changes a method to
+   * {@code final} and a subclass is still overriding the method. Another example is when there is a
+   * method call to an interface and the interface is changed to a class with the same name.
+   *
+   * @see <a href="https://docs.oracle.com/javase/specs/jls/se7/html/jls-13.html#jls-13.4.9">Java
```
Comment (netdpb) [inline dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ErrorType.java]: Should all the JLS references be to the same version? (This one is Java 7, and the next is Java 8.)

Which version?

File: dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/SymbolProblem.java [hunk 4]
```diff
@@ -76,20 +76,20 @@ public boolean equals(Object other) {
...
+    return Objects.hash(errorType, symbol, containingClass);
   }
 
   @Override
   public String toString() {
     return MoreObjects.toStringHelper(this)
         .add("errorType", errorType)
         .add("symbol", symbol)
-        .add("targetClass", targetClass)
+        .add("containingClass", containingClass)
```
Comment (netdpb) [inline dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/SymbolProblem.java]: Maybe also call `.omitNullValues()`?

File: dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ErrorType.java (hunk 3 above)
Comment (suztomo) [reply dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ErrorType.java]: Updated to Java 8.

File: dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/SymbolProblem.java (hunk 4 above)
Comment (suztomo) [reply dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/SymbolProblem.java]: Added the method. Good to know that. Thanks.

**Pull request merged**